File: ios/AgoraRtcUniPlugin/Uni/UniCallback.swift
//
//  UniCallback.swift
//  AgoraRtcUniPlugin
//
//  Created by LXH on 2020/12/14.
//

import Foundation
import AgoraRtcKit

@objc(UniCallback)
class UniCallback: NSObject, Callback {
    private var callback: WXModuleCallback?

    init(_ callback: WXModuleCallback?) {
        self.callback = callback
    }

    func success(_ data: Any?) {
        callback?(data)
    }

    func failure(_ code: String, _ message: String) {
        callback?(["code":code, "message":message])
    }
}

File: android/src/main/java/io/agora/rtc/uni/UniCallback.kt
package io.agora.rtc.uni

import com.alibaba.fastjson.JSONObject
import io.agora.rtc.base.Callback
import io.dcloud.feature.uniapp.bridge.UniJSCallback

class UniCallback(
        private val callback: UniJSCallback?
) : Callback() {
    override fun success(data: Any?) {
        callback?.invoke(data)
    }

    override fun failure(code: String, message: String) {
        callback?.invoke(JSONObject().apply {
            put("code", code)
            put("message", message)
        })
    }
}
